src/input.rs

```diff
@@ -16,83 +21,80 @@ pub(crate) struct App {
 }
 
 impl App {
-    fn stdin_replace(&self, is_tty: bool) -> Result<()> {
-        let mut buffer = Vec::with_capacity(256);
-        let stdin = std::io::stdin();
-        let mut handle = stdin.lock();
-        handle.read_to_end(&mut buffer)?;
-
-        let stdout = std::io::stdout();
-        let mut handle = stdout.lock();
-
-        handle.write_all(&if is_tty {
-            self.replacer.replace_preview(&buffer)
-        } else {
-            self.replacer.replace(&buffer)
-        })?;
-
-        Ok(())
-    }
-
     pub(crate) fn new(source: Source, replacer: Replacer) -> Self {
         Self { source, replacer }
     }
+
     pub(crate) fn run(&self, preview: bool) -> Result<()> {
-        let is_tty = std::io::stdout().is_terminal();
+        let sources: Vec<(PathBuf, Mmap)> = match &self.source {
+            Source::Stdin => {
+				let mut handle = stdin().lock();
+				let mut buf = Vec::new();
+				handle.read_to_end(&mut buf)?;
+				let mut mmap = MmapOptions::new()
+					.len(buf.len())
+					.map_anon()?;
+				mmap.copy_from_slice(&buf);
+				let mmap = mmap.make_read_only()?;
+				vec![(PathBuf::from("STDIN"), mmap)]
+			},
+            Source::Files(paths) => {
+				let mut refs = Vec::new();
+                for path in paths {
+                    if !path.exists() {
+                        return Err(Error::InvalidPath(path.clone()));
+                    }
+					let mmap = unsafe { Mmap::map(&File::open(path)?)? };
+                    refs.push((path.clone(), mmap));
+                }
+                refs
+            },
```

Comment:
This is behavior that should have a test to verify, so that we don't have to manually keep track of things, but this changes the behavior of how `sd` handles failed jobs (in terms of files not exist, not being readable, etc.) where it now stops of the first failure here instead of processing all files that could be handled while reporting failures on the others

I think I prefer keeping with the current behavior where we still process the files that we can while reporting the ones that failed

Reply:
Sure, can go with `Vec<Result<>>`. Then it would warrant a reordering of display I think, putting error cases at the end, otherwise they might easily slip away.

Reply:
Sounds good to me!